spyvm/plugins/fileplugin.py

```diff
@@ -167,19 +186,45 @@ def primitiveFileStdioHandles(interp, s_frame, w_rcvr):
     space = interp.space
     return space.wrap_list([space.wrap_int(fd) for fd in std_fds])
 
-@FilePlugin.expose_primitive(unwrap_spec=[object, int, str, index1_0, int])
-def primitiveFileWrite(interp, s_frame, w_rcvr, fd, a_string, start, count):
+@FilePlugin.expose_primitive(unwrap_spec=[object, int, object, index1_0, int])
+def primitiveFileWrite(interp, s_frame, w_rcvr, fd, content, start, count):
+    byte_size = 0
+    if isinstance(content, model.W_WordsObject):
+        byte_size = 4
+    elif isinstance(content, model.W_BytesObject):
+        byte_size = 1
+    else:
+        raise PrimitiveFailedError
+
+    string_content = content.as_string()
+    byte_start = start * byte_size
+    byte_end = min(start + 1 + count, content.size()) * byte_size
+
     space = interp.space
-    end = min(start + 1 + count, len(a_string))
-    if not (start >= 0 and end > start):
+
+    if not (byte_start >= 0 and byte_end > byte_start):
         return space.wrap_int(0)
     try:
-        written = os.write(fd, a_string[start:end])
+        written = os.write(fd, string_content[byte_start:byte_end])
     except OSError:
         raise PrimitiveFailedError
     else:
         return space.wrap_positive_32bit_int(rarithmetic.intmask(written))
 
+@FilePlugin.expose_primitive(unwrap_spec=[object, int, int])
+def primitiveFileTruncate(interp, s_frame, w_rcvr, fd, position):
+    try:
+        # TODO: ftruncate might not be available on Windows
+        os.ftruncate(fd, position)
+    except OSError:
+        raise PrimitiveFailedError
+    return w_rcvr
+
+@FilePlugin.expose_primitive(unwrap_spec=[object, str, str, str])
+def primitiveDirectorySetMacTypeAndCreator(interp, s_frame, w_rcvr, filename, type, creator):
+    # TODO: this is a stub. "MacOS.SetCreatorAndType" is not available in my pypy build
+    return w_rcvr
```

Comment:
We should just do nothing and ok. Mac OS creator types are completely useless nowadays